packages/client/src/tx/create-tx.ts

```diff
@@ -90,3 +106,64 @@ export const createTx: (
       )
     }),
   )
+
+const getNonce$ = (chainHead: ChainHead$, from: HexString) => {
+  const followHead$ = (head: string) =>
+    chainHead.newBlocks$.pipe(
+      scan((acc, block) => (block.parent === acc ? block.hash : acc), head),
+      startWith(head),
+      distinctUntilChanged(),
+    )
+  const followNonce$ = (head: string) =>
+    followHead$(head).pipe(
+      switchMap((hash) => getNonceAtBlock$(chainHead.call$, from, hash)),
```

Comment:
I'm unsure how this will behave in flawed situations (slow conn, bad CPU, smoldot, whatever). Even more so for fast chains like Aleph Zero, with blocks appearing every second, I'm afraid this can affect and take longer just because ops are being cancelled and restarted every time.

I'd rather take a snapshot of the blocks at the beginning of the whole nonce retrieval, or limit the number of blocks that this can stop the emission.

Reply:
I'd like to cover the case where we have a bestBlock, and just as we were waiting for a nonce value, a new block came in with a new tx.

My only concern is with the case of fast chains... something that could be done is only do this for the first block coming through for each head, if new ones keep cancelling old nonce requests then ignore them.

It is extremely unlikely to happen, so I'm ok dropping this logic altogether, but I feel like it could help?

Reply:
Yep, I think it can help, but let's just limit the number of new blocks that will be used. I just just a `take(n)` before the `switchMap` will do the job.